apps/frontend/src/app/poll/choose-events/choose-events.component.ts

```diff
@@ -202,13 +206,23 @@ export class ChooseEventsComponent implements OnInit {
     } else if (maxParticipants && this.participants.length >= maxParticipants) {
       this.closedReason = 'This poll has reached it\'s maximum number of participants. You can no longer submit your vote.';
       this.showResults = true;
+    } else if (this.poll?.settings?.showResult === ShowResultOptions.NEVER as ShowResultOptions) {
+      this.closedReason = undefined;
+      this.showNever = !this.userVoted() && !this.isAdmin;
+      if (!this.isAdmin) {
+        this.participants = this.participants.filter(p => p.token === this.token);
+      }
+      this.showResults = true;
     } else {
       this.closedReason = undefined;
-      this.showResults = !this.poll?.settings?.blindParticipation || this.isAdmin || this.userVoted();
+      this.showAfterParticipating = !this.userVoted() && !this.isAdmin &&
+        this.poll?.settings?.showResult === ShowResultOptions.AFTER_PARTICIPATING;
+      this.showResults = this.isAdmin || this.userVoted() ||
+        this.poll?.settings?.showResult === ShowResultOptions.IMMEDIATELY;
```

Comment:
Can these boolean variables not be simplified to a check in the template?

Reply:
`showAfterParticipating` was moved to template: 9365424. I think we can leave `showResults` as it was previously (just noticed that my commit message is wrong tho, whoopsie).